### ISSUE TITLE
rules-for-equality  -- Clarifying what paths/identification might be, and how path induction constrains us.

### ISSUE THREAD

**Issue by chaikens** [UniMath/SymmetryBook#124]
When trying to understand the path/identification/equality types, I'm looking for examples of what they can, and also CANNOT be.  It's gratifying to read the example in Euclidean geometry of congruence symmetries as paths in the triangle type.  However, this leads me to wonder about two questions:

(1) Does ML type theory forbid us to then allow non-congruence preserving transformations, like affine scaling, non-uniform scaling, skew, etc. transformations to be considered functions?  

(2) Could we take the paths in the triangle type be similarities?  These are not preserved by general affine transformations either.

In general, would it make sense to discuss how (whether?) the path induction principle CONSTRAINS what could be types and functions?  I wonder about this and haven't found a discussion in terms of constraints in the HoTT literature.  I would appreciate NON-EXAMPLES in addition to the whole range of elementary examples and applications.

Thanks, and I would appreciate your patience with my naivety. 
Seth

**Comment by UlrikBuchholtz** [comment]
These are great questions, and we hope to shed a lot of light on them later in the book, though the relevant parts are mostly yet to be written.

But in brief: The key is to consider different types of planes. There are euclidean planes (with or with orientation), affine planes, “similarity planes”, etc., and they each have different symmetries, though they all have underlying sets of points that are abstractly (merely) isomorphic. A euclidean triangle consists of a euclidean plane equipped with a subset of three points in general position, so an identification of such consists of an identification of euclidean planes that preserves the subsets. There are forgetful maps, so, e.g., we can map euclidean planes to affine planes, and hence euclidean triangles to affine triangles.

And there are more complicated relationships when we consider projective geometry or hyperbolic geometry, for instance, where there are several possible notions of “underlying set”.

**Comment by chaikens** [comment]
Hi Ulrik,
Thanks for your prompt answer!

So can I say the only symmetries of the Euclidean plane with orientation are congruences, including
those that reverse orientation?   So the only paths in Euclidean plane types are congruences?

Seth
________________________________________

**Comment by DanGrayson** [comment]
Yes, Euclidean planes are equipped with a function that gives the distance between any two points, so symmetries of the Euclidean plane must preserve distance.  Projective planes are not, so symmetries of projective planes are different.

**Comment by DanGrayson** [comment]
When you say you are considering "symmetries of the Euclidean plane with orientation", then the symmetries must preserve the orientation, as that is part of the structure.

**Comment by chaikens** [comment]
So, again, the possible paths in a Euclidean plane type are all and only the Euclidean congruences,
orientation preserving for the oriented plane and general for the unoriented plane?

Now, is the above assertion something we can prove using path induction?  In other words,
how do we justify the above assertion?

Seth

________________________________________

**Comment by UlrikBuchholtz** [comment]
Once we set up the types involved (here, the type of oriented euclidean planes), then we prove the assertion that the paths in this type correspond to orientation preserving motions by an analysis of the type: It's going to be a Σ-type with the underlying set as the first component, so any path has to contain a bijection on the underlying sets, then the further components constrain the bijection until it's forced to be an orientation preserving congruence. So it's not directly by path induction, although that's involving in analyzing paths in Σ-types.

**Comment by DanGrayson** [comment]
Yes, that's right.

Here's a similar assertion with a proof that appeals to univalence.  The proof also uses the identification of a path
between pairs with a pair of paths -- perhaps that should be clarified.

<img width="716" alt="Screen Shot 2021-09-12 at 3 44 02 PM" src="https://user-images.githubusercontent.com/700228/133000660-86cc71ca-42ee-4469-a70c-5cc3b83cc9eb.png">

The HoTT book treats this more abstractly, in section 9.8 on the structure identity principle.
